clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1073,14 +1078,16 @@ static void checkExprLifetimeImpl(Sema &SemaRef,
     }
 
     case LK_Assignment: {
-      if (!MTE)
+      if (!MTE || pathContainsInit(Path))
         return false;
       assert(shouldLifetimeExtendThroughPath(Path) ==
                  PathLifetimeKind::NoExtend &&
              "No lifetime extension for assignments");
-      if (!pathContainsInit(Path))
-        SemaRef.Diag(DiagLoc, diag::warn_dangling_pointer_assignment)
-            << AEntity->LHS << DiagRange;
+      SemaRef.Diag(DiagLoc,
+                   IsGslPtrInitWithGslTempOwner
```

Comment:
`IsGslPtrInitWithGslTempOwner` might no longer be an accurate name as it is also used for assignments. Naming is hard, I am ok leaving this as is. Alternatively, it could be something like `IsGslPtrValueFromGslTempOwner`.

Reply:
Done, renamed to `IsGslPtrValueFromGslTempOwner`.